expression/vectorized.go

```diff
@@ -21,8 +21,7 @@ import (
 )
 
 func genVecFromConstExpr(ctx sessionctx.Context, expr Expression, input *chunk.Chunk, result *chunk.Column) error {
-	n := input.NumEffectiveRows()
-	sel := input.Sel()
+	n := input.NumRows()
```

Comment:
The `Sel` of input is considered here.
Since the result now is dense, just putting `n` values into the result column is enough.